disnake/client.py

```diff
@@ -405,23 +358,25 @@ def __init__(
         # self.ws is set in the connect method
         self.ws: DiscordWebSocket = None  # type: ignore
 
-        if loop is None:
-            with warnings.catch_warnings():
-                warnings.simplefilter("ignore", DeprecationWarning)
-                self.loop: asyncio.AbstractEventLoop = asyncio.get_event_loop()
-        else:
-            self.loop: asyncio.AbstractEventLoop = loop
-
-        self.loop.set_debug(asyncio_debug)
         self._listeners: Dict[str, List[Tuple[asyncio.Future, Callable[..., bool]]]] = {}
         self.session_start_limit: Optional[SessionStartLimit] = None
 
+        if connector:
+            try:
+                asyncio.get_running_loop()
+            except RuntimeError:
+                raise RuntimeError(
+                    (
+                        "`connector` was created outside of an asyncio loop; consider moving bot class "
+                        "instantiation to an async main function and then manually asyncio.run it"
```

Comment:
TODO: improve this message to explain why this is an error (it is an error because connectors when instantiated create a custom loop if there is no existing running, which leads to errors due to loop mismatch between the one that bot uses and the one that the connector uses)